drivers/sensor/iis2mdc/iis2mdc.c

```diff
@@ -367,6 +336,119 @@ static int iis2mdc_init(const struct device *dev)
 	return 0;
 }
 
-DEVICE_DT_INST_DEFINE(0, iis2mdc_init, device_pm_control_nop,
-		     &iis2mdc_data, &iis2mdc_dev_config, POST_KERNEL,
-		     CONFIG_SENSOR_INIT_PRIORITY, &iis2mdc_driver_api);
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "IIS2MDC driver enabled without any devices"
```

Comment:
This happens to all Zephyr drivers, but I haven't seen such kind of warning. I think it should be discussed with maintainers, if found useful it could be made available somehow to all drivers.

Reply:
> I think it should be discussed with maintainers, if found useful it could be made available somehow to all drivers.

I think it's nicer to proceed incrementally or perhaps treat this as a quality of implementation issue. There are lots of inconsistencies in how we emit warnings and log issues in drivers and I'm not sure it is a good idea to hold up this PR on decisions for how to standardize.